tests/pester-tests/commands/choco-search.Tests.ps1

```diff
@@ -488,6 +488,380 @@ Describe "choco <_>" -ForEach $Command -Tag Chocolatey, SearchCommand, FindComma
+                $Output.Lines | Should -Contain "3 packages found." -Because $Output.String
+            }
+        }
+
+        Context "Searching for the word 'package' with --by-tags-only" {
+            BeforeAll {
+                $Output = Invoke-Choco $_ Package --by-tags-only
+            }
+
+            It "Exits with Success (0)" {
+                $Output.ExitCode | Should -Be 0 -Because $Output.String
+            }
+
+            It "Should contain the following packages" {
+                $Output.Lines | Should -Contain "msi.template 1.0.2" -Because $Output.String
+                $Output.Lines | Should -Contain "zip.template 1.0.0" -Because $Output.String
+            }
+
+            It "Should contain a summary with 2 package versions" {
+                $Output.Lines | Should -Contain "2 packages found." -Because $Output.String
+            }
+        }
+
+        Context "Doing an open-ended search with no ordering but limit-output" {
+            BeforeAll {
+                $Output = Invoke-Choco $_ --limit-output
+            }
+
+            It "Exits with Success (0)" {
+                $Output.ExitCode | Should -Be 0 -Because $Output.String
+            }
+
+            It "Should have add-path as first package" {
+                $Output.Lines[0] | Should -Contain "add-path|1.0.0" -Because $Output.String
+            }
+
+            It "Should contain a summary with 71 package versions" {
+                $Output.Lines | Should -Contain "71 packages found." -Because $Output.String
+            }
+        }
+
+        Context "Doing an open-ended search with --order-by-popularity and limit-output" {
+            BeforeAll {
+                $Output = Invoke-Choco $_ --limit-output --order-by-popularity
+            }
+
+            It "Exits with Success (0)" {
+                $Output.ExitCode | Should -Be 0 -Because $Output.String
+            }
+
+            It "Should have package13 as first package" {
+                # There is some additional output
+                # about the deprecation of
+                # --order-by-popularity so we need to
+                # actually grab the 3rd line
+                $Output.Lines[0] | Should -Contain "'--order-by-popularity' is deprecated and will be removed in a future release." -Because $Output.String
+                $Output.Lines[1] | Should -Contain " Use '--order-by='Popularity'' instead." -Because $Output.String
+                $Output.Lines[2] | Should -Contain "package13|10.10.10" -Because $Output.String
+            }
+
+            It "Should contain a summary with 71 package versions" {
+                $Output.Lines | Should -Contain "71 packages found." -Because $Output.String
+            }
+        }
+
+        Context "Doing an open-ended search with --order-by=popularity and limit-output" {
+            BeforeAll {
+                $Output = Invoke-Choco $_ --limit-output --order-by=popularity

```

Comment:
I think I've answered this myself, but will add it here: is this intentionally `--order-by=popularity` and not `--order-by-popularity`? Assuming that it is, we should probably add a comment indicating that the `=` is intentional (or perhaps `--order-by="'popularity'"`)

Reply:
Yes, that is correct.  `--order-by-popularity` is now marked as obsolete, and the recommendation is to use `--order-by=` which was added in a recent version of Chocolatey CLI.  It has an enumeration for the values that _can_ be sorted by, i.e. title, id, lastpublished, popularity, etc.  I will get the usage of this option updated based on the current documentation for this option.

Reply:
I would agree with doing the single quotes. When I first came across this line, I mistook the `=` for intending to be `-`. But by using single quotes I think it makes it more clear that it was intentional.

Reply:
I've updated all instances of `--order-by=[^']` to be the quoted variant.